### PR TITLE
Support snowflake and bq

### PR DESCRIPTION
**Pull request by drewbanin**
This PR rounds out support for Snowflake and BigQuery. Additionally, this PR adds a nontrivial test suite for Redshift, Snowflake, and BigQuery.

A number of date dimensions were removed from the `page_views` and `sessions` models in order to facilitate cross-database support. These dimensions were largely duplicative in nature. Besides these deletions, there should be no material changes to the Redshift or Snowflake versions of these models.

The BigQuery models included here work differently than the Snowflake and Redshift models. They start with a table of events which are rolled up to pageviews, then sessions, then users. In each rollup, the relevant events from the preceding model are preserved in a repeated record.

TODO:
 - [x] Implement models using `adapter_macro` for Snowflake, Redshift, and BQ
 - [x] Add integration tests
 - [x] Make integration tests run in Circle on PRs
 - [x] Remove "bounce" column from pageviews
 - [x] investigate browser/useragent fields in sessions table (`snowplow_web_ua_parser_context`?)

### CODE REVIEW

**Comment by drewbanin** [comment]
I just removed the `bounce` metric from `page_views`, and added the OS information present in core events. To get rich OS/Device info, Snowplow users should enable the [ua-parser-enrichment](https://github.com/snowplow/snowplow/wiki/ua-parser-enrichment) and enable the corresponding models in this package.

I think we can merge this!

**Review by jthandy — COMMENTED** [review]
One question: there is a real test failure on RS from what I can see. Shouldn't we address that before merging?

Otherwise I'm good to merge whenever...

**Comment by drewbanin** [comment]
Thanks @jthandy - good catch. Just pushed some changes... can merge this when the tests pass

**Review by jthandy — APPROVED** [review]

**Pull request merged**
